### DISCUSSION TITLE
Not possible to use Guard Conditions with References?

### DISCUSSION THREAD

**Discussion by trusktr** [eclipse-langium/langium#1046] [Q&A]
It does not seem that this is allowed:

```js
Expr:
  foo=[GuardedThing<true>:ID]
```

Do I need to make reference to each version of the guard?

**Comment by msujew** [comment]
During the runtime of Langium, the scoping mechanism is not aware which element was parsed with a specific set of guard conditions. These guards can only be used to influence parsing decisions, and nothing else.

Note that a reference references an AST object type, and not a parser rule. The default behavior for parser rules is just that they generate a type named after themselves.
